bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/ConfigUtil.java

```diff
@@ -63,9 +66,51 @@ public static Object normalizeType(Object value, ConfigDescriptionParameter conf
         if (configDescriptionParameter != null) {
             Normalizer normalizer = NormalizerFactory.getNormalizer(configDescriptionParameter);
             return normalizer.normalize(value);
-        } else {
-            return value instanceof Double ? BigDecimal.valueOf((Double) value) : value;
+        } else if (value == null || value instanceof Boolean || value instanceof String
+                || value instanceof BigDecimal) {
+            return value;
+        } else if (value instanceof Number) {
+            return new BigDecimal(value.toString());
+        } else if (value instanceof Collection) {
+            return normalizeCollection((List) value);
```

Comment:
If you check for `Collection` wouldn't it be better to work on the collection interface (and perhaps create a new list) instead of rely that the collection will always be a list on this place (if it is the case, then use `instanceof List`.